scripts/file_collector.py

```diff
@@ -0,0 +1,187 @@
+            if name not in variables.keys():
+                variables[name] = values
+
+    # get values for all variables we would need if we were doing an evaluation
+    logger.info("Collecting variables")
+    Session = sessionmaker(create_engine(connection_string))
+    sesh = Session()
+
+    # write file paths for each variable to text file
+    for name, values in variables.items():
+        write_file_paths(sesh, values, ensemble, date_range, region)
+
+
+def write_file_paths(sesh, variables, ensemble, date_range, area):
+    """Given a variable name write the required file's path to file_paths.txt
+    by querying the CE backend.
+    """
+    logger.info("")
+    logger.info("Translating variables for query")
+    query_args = translate_args(
+        variables["variable"],
+        variables["time_of_year"],
+        variables["temporal"],
+        variables["spatial"],
+        variables["percentile"],
+        area,
+        date_range,
+        ensemble,
+    )
+
+    logger.info("Collecting models")
+    models = get_models(sesh, variables["percentile"], ensemble)
+
+    var_name = "_".join(
+        [
+            variables["variable"],
+            variables["time_of_year"],
+            variables["temporal"],
+            variables["spatial"],
+            variables["percentile"],
+        ]
+    )
+
+    logger.info("Fetching data for {}".format(var_name))
+
+    paths = []
+    [paths.extend(query_backend(sesh, model, query_args)) for model in models]
+    logger.info("Writing file path for {} to file".format(var_name))
+
+    with open("file_paths.txt", "a") as file_:
+        for path in paths:
+            file_.write(path + "\n")
+
+
+def query_backend(sesh, model, query_args):
+    """Return the desired file name for a particular climate model"""
+    logger.debug("Running query_backend() with args: %s, %s", model, query_args)
+    ids = [
+        search_for_unique_ids(
+            sesh,
+            ensemble_name=query_args["ensemble_name"],
+            model=model,
+            emission=query_args["emission"],
+            time=query_args["time"],
+            variable=var,
+            timescale=query_args["timescale"],
+            cell_method=query_args["cell_method"],
+        )
+        for var in query_args["variable"]
+    ]
+    file_names = []
+    for id_ in ids:
+        for model_id in list(id_):
+            data_file = (
+                sesh.query(DataFile).filter(DataFile.unique_id == model_id).one()
+            )
+            if data_file is not None:
+                file_names.append(data_file.filename)
+    return file_names
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument(
+        "-c", "--csv", help="CSV file containing rules", default="./data/rules.csv",
+    )
+    parser.add_argument(
+        "-d",
+        "--date-range",
+        help="30 year period for data",
+        choices=["2020", "2050", "2080"],
+        default="2080",
+    )
```

Comment:
I had a thought about how this script should work by default. Since the intention is to collect all the files we need we want to ensure we hit all the different time periods. Instead of having the user have to manually run the script for each time period, by default, can we have all of the periods selected and loop over them?